File: src/api/import.ts
import { invoke } from "@tauri-apps/api/core";

export interface CsvImportReport {
  total_rows: number;
  books_created: number;
  books_found: number;
  tags_created: number;
  tag_associations: number;
  errors: string[];
}

/** 批量导入 CSV 格式的图书标签 */
export async function importBooksCsv(csvText: string): Promise<CsvImportReport> {
  return invoke("import_books_csv", { csvText });
}

File: src-tauri/src/commands/import.rs
use crate::models::CsvImportReport;
use crate::services::repository;
use tauri::command;

/// 批量导入 CSV 格式的图书标签
/// 每行格式：图书名称,标签1,标签2,...
/// 图书不存在时自动创建，标签不存在时自动创建
#[command]
pub fn import_books_csv(csv_text: String) -> Result<CsvImportReport, String> {
    repository::import_books_csv(&csv_text)
}
